### PR TITLE
Identify Qiita users by permanent_id

### PR DESCRIPTION
**Pull request by yktakaha4**
## Proposed changes

- Generate UserSocialAuth with `permanent_id` in [Qiita](https://qiita.com/) backend.
  - Currently Qiita backend identifies UserSocialAuth by `id`, but this value can be changed by renaming the account name.
  - According to the [official documentation](https://qiita.com/api/v2/docs#get-apiv2authenticated_user), `permanent_id` is available as a generated ID for each user.
- To maintain compatibility, the behavior is modified by the settings variable `SOCIAL_AUTH_QIITA_IDENTIFIED_BY_PERMANENT_ID`.
  - If unspecified, identified by `id`.
- More user information can be referenced in `extra_data` .
- Add tests for new behaviors.

## Types of changes

- [ ] Release (new release request)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (PEP8, lint, formatting, renaming, etc)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes (build process, tests runner, etc)
- [ ] Other (please describe):

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Other information

I use this library in my business and always appreciate it!
I am not an English speaker, so I would be happy to point out any discrepancies in naming variables, etc.

### CODE REVIEW

**Comment by yktakaha4** [comment]
I verified that the test passed with the `make tests` command in my local, but it did not run for Python 3.10.

image

**Comment by nijel** [comment]
Can you please update the documentation at https://github.com/python-social-auth/social-docs as well?

**Comment by yktakaha4** [comment]
Thank you for your review. I have created the following PR.
https://github.com/python-social-auth/social-docs/pull/116

File: social_core/backends/qiita.py [hunk 1]
```diff
@@ -65,3 +72,16 @@ def user_data(self, access_token, *args, **kwargs):
...
             }
         )
+
+    def get_user_id(self, details, response):
+        """Return user id"""
+        user_id = None
+        if self.setting('IDENTIFIED_BY_PERMANENT_ID'):
+            user_id = response.get('permanent_id')
+        else:
+            user_id =  response.get('id')
```
Comment (nijel) [inline social_core/backends/qiita.py:82]:
```suggestion
            user_id = response.get('id')
```

**Review by nijel — APPROVED** [review]
Looks good, now we need to fix the flake8 errors in the rest of code to allow merging...

**Comment by yktakaha4** [comment]
@nijel I was going to create another branch to handle this, but I messed up using the forked repository and added the commit to this branch.
I am sorry but please point out any problems.

In my local:

```
$ flake8 social_core/      
$ echo $?       
0
```

**Comment by nijel** [comment]
Can you please separate the flake8 changes from the Qiita ones? Keeping it one topic in PR is preferred...

**Comment by yktakaha4** [comment]
My apologies. I have created a flake8 PR based on the current master branch.
https://github.com/python-social-auth/social-core/pull/707

Also, the flake8 error related to qiita has been fixed in this PR.
https://github.com/python-social-auth/social-core/pull/698/commits/b5b63958053bf7e6f6b1f6952c07308dc291bd16

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**